### ISSUE TITLE
Request: allow restoring state of rotation without storing containerTransformMatrix

### ISSUE THREAD

**Issue by tarngerine** [ailon/markerjs2#118]
There is a private function `applyRotation` inside `RectangularBoxMarkerBase` which is only run when the user actively uses the cursor to manipulate the shape
https://github.com/ailon/markerjs2/blob/e9a0b3bb8fde5fba355e737047ee4819bddc8a58/src/markers/RectangularBoxMarkerBase.ts#L345-L349

I'd like to request that we run `applyRotation` on `restoreState` as well — this means that someone (like me :p ) who is not storing `containerTransformMatrix` can just store `rotationAngle` and the marker can restore its initial rotation angle without user interaction

I have tested this on my own local class extending RectangularBoxMarkerBase and it seems to work well. Obviously that means I am not blocked by this request — it's just something I observed as might be nice for the system overall :) and I don't have to extend every Marker class to add this behavior. 
<img width="805" alt="image" src="https://user-images.githubusercontent.com/1629901/186263495-6aa3d744-2ae5-4d74-ba69-9a25dfc239d4.png">

**Comment by ailon** [comment]
I'd need to dig deep into my memory (or rather git history ;) to remember why the matrices were actually needed. `applyRotation()` was there when those matrices were introduced, so there must have been a reason. I'd be really afraid of just swapping one for the other and forgetting some edge case that breaks.

**Comment by tarngerine** [comment]
You're right this is tricky, turns out you can't just apply it at the end of restoreState. Instead it should be when restoringState but after scale is called, since this.centerX/centerY are different after scaling. Renaming this issue to clarify the intent

**Comment by tarngerine** [comment]
Totally fair! I’ll just post up what I find in case it’s useful :) thanks for entertaining all my issues lol
